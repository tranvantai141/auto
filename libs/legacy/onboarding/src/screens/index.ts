export { default as FirstScreen } from "@src/screens/Auth.FirstScreen/View.FirstScreen";
