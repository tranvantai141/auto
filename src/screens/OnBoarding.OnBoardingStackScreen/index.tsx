import { createStackNavigator, TransitionPresets } from "@react-navigation/stack";
import { EOnBoardingScreenList } from "@src/models/RouterNamesModel";
import React from "react";
import StepOneScreen from "../OnBoarding.StepOneScreen/View.StepOneScreen";
import StepTwoScreen from "../OnBoarding.StepTwoScreen/View.StepTwoScreen";

export type TOnBoardingStack = {
  [EOnBoardingScreenList.ONBOARDING_STEP_ONE_SCREEN]: undefined;
  [EOnBoardingScreenList.ONBOARDING_STEP_TWO_SCREEN]: undefined;
};

const Stack = createStackNavigator<TOnBoardingStack>();

const OnBoardingStackScreen = React.memo(() => {
  return (
    <Stack.Navigator
      initialRouteName={EOnBoardingScreenList.ONBOARDING_STEP_ONE_SCREEN}
      screenOptions={{
        headerShown: false,
      }}
    >
      <Stack.Screen
        name={EOnBoardingScreenList.ONBOARDING_STEP_ONE_SCREEN}
        options={TransitionPresets.ModalSlideFromBottomIOS}
        component={StepOneScreen}
      />

      <Stack.Screen
        name={EOnBoardingScreenList.ONBOARDING_STEP_TWO_SCREEN}
        options={TransitionPresets.ModalSlideFromBottomIOS}
        component={StepTwoScreen}
      />
    </Stack.Navigator>
  );
});

OnBoardingStackScreen.displayName = "OnBoardingStackScreen";
export default OnBoardingStackScreen;
