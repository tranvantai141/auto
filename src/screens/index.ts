export { default as FirstScreen } from "@src/screens/Auth.FirstScreen/View.FirstScreen";
export { default as LoginScreen } from "@src/screens/Auth.LoginScreen/View.LoginScreen";
export { default as StepOneScreen } from "@src/screens/OnBoarding.StepOneScreen/View.StepOneScreen";
export { default as StepTwoScreen } from "@src/screens/OnBoarding.StepTwoScreen/View.StepTwoScreen";
export { default as OnBoardingStackScreen } from "@src/screens/OnBoarding.OnBoardingStackScreen/index";
