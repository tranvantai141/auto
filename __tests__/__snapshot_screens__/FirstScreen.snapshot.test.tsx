import { FirstScreen } from "@src/screens";
import renderer from "react-test-renderer";

jest.mock("react-native-date-picker", () => "DatePicker");

it("FirstScreen renders correctly", () => {
  const tree = renderer.create(<FirstScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});
