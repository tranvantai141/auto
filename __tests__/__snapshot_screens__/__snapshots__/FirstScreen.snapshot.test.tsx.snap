// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FirstScreen renders correctly 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <Image
    accessible={false}
    resizeMode="stretch"
    source={
      Object {
        "testUri": "../../../src/assets/images/LoginBackground.jpg",
      }
    }
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": undefined,
          "width": undefined,
        },
        undefined,
      ]
    }
    testID="FirstScreen-imageBackground"
  />
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "height": 180.7142857142857,
          "justifyContent": "center",
          "width": "100%",
        }
      }
    >
      <View
        accessible={false}
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "space-between",
            "marginTop": 20,
          }
        }
        testID="OnboardingComponent-1container"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "aspectRatio": 1,
              "backgroundColor": "#0FDE82",
              "borderRadius": 100,
              "height": 57.5,
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#ffffff",
                "fontFamily": "Inter-Regular",
                "fontSize": 9.75,
              }
            }
          >
            1
          </Text>
        </View>
        <Text
          style={
            Object {
              "color": "#0FDE82",
              "fontFamily": "Inter-Regular",
              "fontSize": 9.75,
            }
          }
        >
          Basic Info
        </Text>
      </View>
      <View
        accessible={false}
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "space-between",
            "marginTop": 20,
          }
        }
        testID="OnboardingComponent-2container"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "aspectRatio": 1,
              "backgroundColor": "#E5E5E5",
              "borderRadius": 100,
              "height": 57.5,
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#336CFB",
                "fontFamily": "Inter-Regular",
                "fontSize": 9.75,
              }
            }
          >
            2
          </Text>
        </View>
        <Text
          style={
            Object {
              "color": "#E5E5E5",
              "fontFamily": "Inter-Regular",
              "fontSize": 9.75,
            }
          }
        >
          Additional Info
        </Text>
      </View>
      <View
        accessible={false}
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "space-between",
            "marginTop": 20,
          }
        }
        testID="OnboardingComponent-3container"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "aspectRatio": 1,
              "backgroundColor": "#E5E5E5",
              "borderRadius": 100,
              "height": 57.5,
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#336CFB",
                "fontFamily": "Inter-Regular",
                "fontSize": 9.75,
              }
            }
          >
            3
          </Text>
        </View>
        <Text
          style={
            Object {
              "color": "#E5E5E5",
              "fontFamily": "Inter-Regular",
              "fontSize": 9.75,
            }
          }
        >
          Purposes
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "marginTop": 32,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#ffffff",
            "fontFamily": "Inter-SemiBold",
            "fontSize": 11.25,
            "marginBottom": 13.142857142857142,
            "marginTop": 26.285714285714285,
          }
        }
      >
        Full-name
      </Text>
      <TextInput
        errorMessageText="Invalid name"
        onBlur={[Function]}
        onChangeText={[Function]}
        placeHolder="Enter full-name"
        placeholder="Enter full-name"
        placeholderTextColor="#9f9f9f"
        setValue={[Function]}
        style={
          Object {
            "alignSelf": "center",
            "backgroundColor": "#E5E5E5",
            "borderColor": "#e4e4e4",
            "borderRadius": 8,
            "borderWidth": 2,
            "color": "#333333",
            "fontFamily": "Inter-Regular",
            "fontSize": 9.75,
            "height": 78.85714285714286,
            "paddingLeft": 16,
            "width": 686,
          }
        }
        testID="username-textinput"
        testId="username-textinput"
        value=""
      />
      <Text
        style={
          Object {
            "color": "#ffffff",
            "fontFamily": "Inter-SemiBold",
            "fontSize": 11.25,
            "marginBottom": 13.142857142857142,
            "marginTop": 26.285714285714285,
          }
        }
      >
        ID number
      </Text>
      <TextInput
        errorMessageText="ID contains 12 digits"
        keyboardType="decimal-pad"
        onBlur={[Function]}
        onChangeText={[Function]}
        placeHolder="Enter 12 digits ID number"
        placeholder="Enter 12 digits ID number"
        placeholderTextColor="#9f9f9f"
        setValue={[Function]}
        style={
          Object {
            "alignSelf": "center",
            "backgroundColor": "#E5E5E5",
            "borderColor": "#e4e4e4",
            "borderRadius": 8,
            "borderWidth": 2,
            "color": "#333333",
            "fontFamily": "Inter-Regular",
            "fontSize": 9.75,
            "height": 78.85714285714286,
            "paddingLeft": 16,
            "width": 686,
          }
        }
        testID="id-textinput"
        testId="id-textinput"
        value=""
      />
    </View>
    <View
      style={
        Object {
          "alignSelf": "center",
          "bottom": 39.42857142857143,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "position": "absolute",
          "width": 710.5714285714286,
        }
      }
    >
      <View
        accessibilityState={
          Object {
            "disabled": true,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "aspectRatio": 1.5,
            "backgroundColor": "#E5E5E5",
            "borderRadius": 100,
            "elevation": 5,
            "height": 72.28571428571428,
            "justifyContent": "center",
            "opacity": 0,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.25,
            "shadowRadius": 3.84,
          }
        }
      >
        <Text
          style={
            Object {
              "fontFamily": "Inter-SemiBold",
              "fontSize": 16.5,
            }
          }
        >
          ←
        </Text>
      </View>
      <View
        accessibilityState={
          Object {
            "disabled": true,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "aspectRatio": 1.5,
            "backgroundColor": "#E5E5E5",
            "borderRadius": 100,
            "elevation": 5,
            "height": 72.28571428571428,
            "justifyContent": "center",
            "opacity": 0.5,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.25,
            "shadowRadius": 3.84,
          }
        }
        testID="nextButton"
      >
        <Text
          style={
            Object {
              "fontFamily": "Inter-SemiBold",
              "fontSize": 16.5,
            }
          }
        >
          →
        </Text>
      </View>
    </View>
    <DatePicker
      androidVariant="iosClone"
      cancelText="Cancel"
      confirmText="Confirm"
      date={2008-06-11T09:07:11.052Z}
      locale="us"
      modal={true}
      mode="date"
      onCancel={[Function]}
      onConfirm={[Function]}
      open={false}
      testID="DatePicker"
      title="Select date and Time"
    />
  </View>
</View>
`;
